### PR TITLE
Add access to Node vars from scripting languages

### PR DESCRIPTION
**Pull request by JimMarlowe**
This PR adds access to the Node c++ class vars_ VariantMap from scripting languages.
The access is done completely by strings, but will handle a large subset of Variant types, those being  Int, Bool, Float, Vector2, Vector3, Vector4, Quaternion, String, Buffer, ResourceRef, ResourceRefList, IntRect, IntVector2, Matrix3,Matrix3x4, Matrix4, Double, Color. It is the users responsibility to convert to and from Variant type strings. The added methods are:
void setUserData (const String& name, const String& vartype, const String& value);  // set data
String getUserData (const String& name) const;  // get data
bool isUserData (const String& name) const;   // data exist?
String getUserDataType (const String& name) const;  //what kind of data is it

### CODE REVIEW

File: Source/Atomic/Scene/Node.h [hunk 1]
```diff
@@ -602,6 +602,39 @@ class ATOMIC_API Node : public Animatable
...
     void GetChildrenWithName(PODVector<Node*>& dest, StringHash nameHash, bool recursive = false) const;
 
+    /// Scripting interface to set user data, the data type can be 
+    ///   Int, Bool, Float, Vector2, Vector3, Vector4, Quaternion
+    ///   String, Buffer, ResourceRef, ResourceRefList, IntRect,
+    ///   IntVector2, Matrix3,Matrix3x4, Matrix4, Double, Color
+    /// The data value must be programed in the appropriate string form
+    /// used by the Variant class.
+    void setUserData (const String& name, const String& vartype, const String& value) 
+    {
```
Comment (JoshEngebretson) [inline Source/Atomic/Scene/Node.h]: C++ method naming convention is camel case capitals, so for example ```SetUserData```

We're mixing ```SetVar``` and ```SetUserData``` when referring to the same ```var_```  VariantMap member.  Something like```SetVarFromString``` would be consistent with ```Var``` and ```Variant::FromString```

On the comment, how about, "The data value must be formatted in the Variant type string format"?

File: Source/Atomic/Scene/Node.h [hunk 2]
```diff
@@ -602,6 +602,39 @@ class ATOMIC_API Node : public Animatable
...
+    /// Scripting interface to check if the user data entry exists
+    /// returns true if present, returns false if it is not present.
+    bool isUserData (const String& name) const {
+        return vars_.Contains(name);
+    }
+
+    /// Scripting interface to get user data type as a string
+    /// an empty string is returned if the entry name is not present.
+    String getUserDataType (const String& name) const {
+         if ( isUserData(name) )
```
Comment (JoshEngebretson) [inline Source/Atomic/Scene/Node.h]: This can return a ```const String&``` saving a string copy on return

File: Source/Atomic/Scene/Node.h [hunk 3]
```diff
@@ -602,6 +602,39 @@ class ATOMIC_API Node : public Animatable
...
+    /// used by the Variant class.
+    void setUserData (const String& name, const String& vartype, const String& value) 
+    { 
+        vars_[name].FromString(vartype, value); 
+    }
+
+    /// Scripting interface to get user data as a string.
+    /// an empty string is returned if the entry name is not present.
+    String getUserData (const String& name) const {
+        if ( isUserData(name) )
```
Comment (JoshEngebretson) [inline Source/Atomic/Scene/Node.h]: Maybe, ```String GetVarAsString(const String& name) const {```?

File: Source/Atomic/Scene/Node.h [hunk 4]
```diff
@@ -602,6 +602,39 @@ class ATOMIC_API Node : public Animatable
...
+            return vars_[name]->ToString();
+        return String::EMPTY;
+    }
+
+    /// Scripting interface to check if the user data entry exists
+    /// returns true if present, returns false if it is not present.
+    bool isUserData (const String& name) const {
+        return vars_.Contains(name);
```
Comment (JoshEngebretson) [inline Source/Atomic/Scene/Node.h]: Sensing a theme, ```bool IsVar(const String& name) const```

File: Source/Atomic/Scene/Node.h (hunk 1 above)
Comment (JimMarlowe) [inline Source/Atomic/Scene/Node.h]: the names have been changed.

File: Source/Atomic/Scene/Node.h (hunk 2 above)
Comment (JimMarlowe) [inline Source/Atomic/Scene/Node.h]: I tried returning a const String&, but both the Variant::GetTypeName() or Variant::ToString() are not const correct, so they generate a returning a temporary compiliation warning if I try to return that.

**Comment by JoshEngebretson** [comment]
Thanks, looks great, landing! ✈️

**Pull request merged**